### PR TITLE
gh-89381: Fix invalid signatures of math/cmath.log

### PR DESCRIPTION
**Pull request by skirpichev**
reopened #101070 [per request](https://github.com/python/cpython/pull/101115#issuecomment-1407609419)

* Issue: gh-89381

### CODE REVIEW

**Comment by skirpichev** [comment]
Probably, changes in the math.rst could be reverted too (like [1efcf49](https://github.com/python/cpython/pull/101404/commits/1efcf49f29f98c68b9b2a1f9126c68f9932ca8be) does).

BTW, I would appreciate Raymond's review.

**Comment by mdickinson** [comment]
Yes, I'd suggest losing the `or is None` in the `math.rst`, too. Other than that, this LGTM.

File: Doc/library/math.rst [hunk 1]
```diff
@@ -393,13 +393,12 @@ Power and logarithmic functions
...
-.. function:: log(x[, base])
+.. function:: log(x, base=None)
 
-   With one argument, return the natural logarithm of *x* (to base *e*).
-
-   With two arguments, return the logarithm of *x* to the given *base*,
-   calculated as ``log(x)/log(base)``.
+   Return the logarithm of *x* to the given *base*.
 
+   If the *base* is not specified or is None, returns the natural
```
Comment (mdickinson) [inline Doc/library/math.rst]: Suggest dropping the `or is None`, for readability - it's clear from context that a value of `None` would be fall under the "not specified" rules.
```suggestion
   If the *base* is not specified, returns the natural
```

**Review by mdickinson — APPROVED** [review]
LGTM

**Comment by skirpichev** [comment]
Thanks for review and I'm sorry for the closed pr.

**Review by hauntsaninja — APPROVED** [review]
Thank you for sticking with this!

**Comment by skirpichev** [comment]
Restored in #101593

**Pull request merged**